content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,171 @@
+This synchronizes the state and the UI, leading to more straightforward and maintainable code.
+
+3. **Consistent user experience**: With the UI always being in sync with the underlying data, users get a consistent and expected response to their actions.
+
+4. **Scalability**: As applications grow, managing state and UI can become complex. Reactivity abstracts this complexity, ensuring that even large-scale applications remain manageable.
+
+## Reactive principles
+
+When delving into the world of reactivity, you'll encounter a few foundational principles:
+
+### Signals
+
+Signals serve as foundational elements in reactive systems, playing a pivotal role in data management and system responsiveness.
+
+A signal can be visualized as a container for data. 
+It not only holds the data, but also provides mechanisms for accessing and modifying its value, a **getter **and a **setter**. 
+This dual capability makes signals both readable and mutable, making them integral to the dynamic nature of reactive systems.
+
+#### Getter
+
+The getter is the mechanism responsible for accessing the current value of the signal. 
+When various components or parts of an application need to "read" or retrieve the data stored in a signal, they use the getter. 
+This provides data that is consistent and reliably accessible throughout the system.
+
+#### Setter
+
+On the other hand, the setter is the tool that permits the changing of the signal's value. 
+Any change to the value within a signal, made possible by the setter, can instigate reactive updates in other connected parts of the system. 
+This ensures that the system remains responsive and up-to-date with the latest data changes.
+
+### Effects
+
+Effects and other listeners (such as memos) can be thought of as automated responders.
+They are closely tied to signals and are set into motion whenever there's a change in the signals they observe. 
+Their primary role is to ensure that any modifications in signals are appropriately reflected throughout the system, maintaining its reactivity.
+
+The nature of effects revolves around two functionalities:
+
+- **Observation**: At the heart of every effect is its ability to continuously monitor or "observe" signals. 
+This keeps effects primed to detect changes in the signals they are tied to.
+- **Activation**: Upon recognizing a change in an observed signal, the effect will promptly execute its predefined routine. 
+This can involve tasks like updating the UI, triggering a process, or initiating external functions. 
+Through this activation, effects ensure that the system's reactivity is both immediate and relevant.
+
+## Building a reactive system
+
+Reactive systems are designed around the principle of data-driven reactivity. This is when signals and their dependents, effects, maintain a relationship. This relationship ensures a seamless propagation of data changes throughout the system, guaranteeing its responsiveness and real-time updates.
```

Comment:
Maybe worth saying "effects and memos", or the reader might think that memos are not dependents of signals.